cupy/testing/time.py

```diff
@@ -0,0 +1,73 @@
+import time
+
+import numpy
+
+import cupy
+
+
+# Borrowed from https://github.com/niboshi/cupy-perf.
+
+class PerfCaseResult(object):
+    def __init__(self, name, ts):
+        self.name = name
+        self.ts = ts
+
+    def cpu_min(self):
+        return self.ts[0].min()
+
+    def cpu_mean(self):
+        return self.ts[0].mean()
+
+    def cpu_std(self):
+        return self.ts[0].std()
+
+    def gpu_min(self):
+        return self.ts[1].min()
+
+    def gpu_mean(self):
+        return self.ts[1].mean()
+
+    def gpu_std(self):
+        return self.ts[1].std()
+
+    def to_str(self, show_gpu=False):
+        s = '{:<20s}: {:9.03f} us   +/-{:6.03f} (min:{:9.03f}) us'.format(
+            self.name,
+            self.cpu_mean() * 1e6,
+            self.cpu_std() * 1e6,
+            self.cpu_min() * 1e6)
+        if show_gpu:
+            s += '  {:9.03f} us   +/-{:6.03f} (min:{:9.03f}) us'.format(
+                self.gpu_mean() * 1e6,
+                self.gpu_std() * 1e6,
+                self.gpu_min() * 1e6)
+        return s
+
+    def __str__(self):
+        return self.to_str(show_gpu=True)
+
+
+def run_perf(name, func, args=(), n=10000, *, n_warmup=10):
+    ts = numpy.empty((2, n,), dtype=numpy.float64)
+    ev1 = cupy.cuda.stream.Event()
+    ev2 = cupy.cuda.stream.Event()
+
+    for i in range(n_warmup):
+        func(*args)
+
+    for i in range(n):
+        ev1.synchronize()
```

Comment:
I don't think you need this synchronization.

Reply:
The GPU time before function entry may be counted if this synchronization is missing.

Reply:
Only the ongoing work in the warmup loop might be counted in this case. How about moving this sync outside the current loop and putting it right after the warmup loop?